repo-agent/issue-sandbox/main.go

```diff
@@ -0,0 +1,185 @@
+
+	// Handle .gemini directory
+	if _, err := os.Stat("/workspaces/.gemini"); err == nil {
+		log.Println(".gemini directory exists in /workspaces, copying to repo directory")
+		if _, err := os.Stat(".gemini"); err == nil {
+			log.Println(".gemini directory exists in repo directory, moving to .gemini.bak")
+			if err := os.Rename(".gemini", ".gemini.bak"); err != nil {
+				return fmt.Errorf("failed to move .gemini to .gemini.bak: %w", err)
+			}
+		}
+		if _, err := runCommand("cp", "-R", "/workspaces/.gemini", ".gemini"); err != nil {
+			return fmt.Errorf("failed to copy .gemini directory: %w", err)
+		}
+	} else {
+		log.Println(".gemini directory does not exist in /workspaces")
+	}
+
+	// Run gemini
+	if _, err := os.Stat("../agent-prompt.txt"); os.IsNotExist(err) {
+		log.Println("agent-prompt.txt does not exist, running gemini")
+		if err := os.WriteFile("../agent-prompt.txt", []byte(agentPrompt), 0644); err != nil {
+			return fmt.Errorf("failed to write agent-prompt.txt: %w", err)
+		}
+		geminiAPIKey := os.Getenv("GEMINI_API_KEY")
+		if geminiAPIKey == "" {
+			geminiAPIKeyBytes, err := os.ReadFile("/tokens/gemini")
+			if err != nil {
+				return fmt.Errorf("failed to read gemini token: %w", err)
+			}
+			geminiAPIKey = string(geminiAPIKeyBytes)
+		}
+		cmd := exec.Command("gemini", "-y", "-p", agentPrompt)
+		cmd.Env = append(os.Environ(), "GEMINI_API_KEY="+geminiAPIKey)
+		output, err := cmd.CombinedOutput()
+		if err != nil {
+			log.Printf("gemini command failed: %v, output: %s", err, string(output))
+		}
+		if err := os.WriteFile("../agent-output.txt", output, 0644); err != nil {
+			return fmt.Errorf("failed to write agent-output.txt: %w", err)
+		}
+	} else {
+		log.Println("agent-prompt.txt exists, skipping gemini generation")
+	}
+
+	// Cleanup .gemini
+	if err := os.RemoveAll(".gemini"); err != nil {
+		log.Printf("failed to remove .gemini directory: %v", err)
+	}
+	if _, err := os.Stat(".gemini.bak"); err == nil {
+		log.Println("moving .gemini.bak -> .gemini")
+		if err := os.Rename(".gemini.bak", ".gemini"); err != nil {
+			return fmt.Errorf("failed to move .gemini.bak to .gemini: %w", err)
+		}
+	}
+
+	// Commit and push
+	if githubUserEmail != "" {
+		if _, err := runCommand("git", "add", "."); err != nil {
+			log.Printf("failed to git add: %v", err)
+		}
+		commitMsg := fmt.Sprintf("fix for issue # %s", issueID)
+		if _, err := runCommand("git", "commit", "-m", commitMsg); err != nil {
+			log.Printf("failed to git commit: %v", err)
+		}
+	}
+
+	newCommitID, err := runCommand("git", "rev-parse", "HEAD")
+	if err != nil {
+		return fmt.Errorf("failed to get new commit id: %w", err)
+	}
+
+	if string(newCommitID) != string(oldCommitID) {
+		log.Println("New changes committed")
+		if gitPushEnabled {
+			if _, err := runCommand("git", "push", "--set-upstream", "origin", issueBranch, "--force"); err != nil {
+				return fmt.Errorf("failed to push changes: %w", err)
+			}
+			log.Println("New changes pushed")
+		}
+	}
```

Comment:
The path to `code-server` is hardcoded. While this is acceptable in a containerized environment where the path is known, it could be more flexible if it were configurable via an environment variable or if the program looked for it in the system's `PATH`.